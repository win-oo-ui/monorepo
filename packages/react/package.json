{
  "name": "@winoo/react",
  "version": "0.1.0",
  "main": "lib/index.js",
  "license": "MIT",
  "files": [
    "lib"
  ],
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose ",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "devDependencies": {
    "@babel/core": "^7.21.0",
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.21.0",
    "@storybook/addon-actions": "^6.5.16",
    "@storybook/addon-essentials": "^6.5.16",
    "@storybook/addon-interactions": "^6.5.16",
    "@storybook/addon-links": "^6.5.16",
    "@storybook/addon-storysource": "^6.5.16",
    "@storybook/builder-webpack5": "^6.5.16",
    "@storybook/manager-webpack5": "^6.5.16",
    "@storybook/preset-typescript": "^3.0.0",
    "@storybook/react": "^6.5.16",
    "@storybook/testing-library": "^0.0.13",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@types/jest": "^29.4.0",
    "@types/react": "^18.0.28",
    "babel-loader": "^8.3.0",
    "jest": "^29.4.3",
    "jest-environment-jsdom": "^29.4.3",
    "react": "^18.2.0",
    "rollup": "^3.18.0",
    "rollup-plugin-typescript2": "^0.34.1",
    "typescript": "^4.9.5"
  },
  "publishConfig": {
    "access": "public"
  }
}
