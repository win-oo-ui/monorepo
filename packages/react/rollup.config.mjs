import Ts from "rollup-plugin-typescript2";

export default {
  input: [
    "src/index.ts",
    // Atoms
    "src/atoms/Button/index.ts",
    "src/atoms/Dot/index.ts",
    "src/atoms/Text/index.ts",
    "src/atoms/Box/index.ts",
    // Molecules
    "src/molecules/Select/index.ts",
    // Organisms
  ],
  output: {
    dir: "lib",
    format: "esm",
    sourcemap: true,
    preserveModules: true,
  },
  plugins: [Ts()],
  external: ["react", "@winoo/foundation"],
};
