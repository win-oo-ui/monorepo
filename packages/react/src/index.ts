export { default as Button } from "./atoms/Button";
export { default as Dot } from "./atoms/Dot";
export { default as Text } from "./atoms/Text";
export { default as Box } from "./atoms/Box";
export { default as Select } from "./molecules/Select";
