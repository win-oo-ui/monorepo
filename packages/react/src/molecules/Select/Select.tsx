import React, {
  KeyboardEvent,
  PropsWithChildren,
  ReactNode,
  createRef,
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState,
} from "react";
import Text from "../../atoms/Text/Text";
import { KeyboardKeys } from "@winoo/foundation";

interface RenderOptionProps<T extends Object> {
  isSelected: boolean;
  isHighlighted: boolean;
  item: T;
  classes: string;
}

interface SelectProps<T extends Object> {
  onSelect?: (options: T, optionIndex: number) => void;
  options: T[];
  keyExtractor?: (value: T) => string;
  labelExtractor: (value: T) => string | number;
  label?: string;
  renderOption?: (props: RenderOptionProps<T>) => ReactNode;
}

const Select = <T extends Object>({
  onSelect,
  options,
  keyExtractor,
  labelExtractor,
  renderOption,
  label = "Please select option...",
}: PropsWithChildren<SelectProps<T>>): JSX.Element => {
  // State Management Of Select Option
  const selectContainerRef = useRef<HTMLDivElement>(null);
  const listContainerRef = useRef<HTMLUListElement>(null);
  const labelRef = useRef<HTMLButtonElement>(null);
  const [optionRefs, setOptionRefs] = useState<
    React.RefObject<HTMLLIElement>[]
  >([]);
  const [selectedIndex, setSelectedIndex] = useState<number>(-1);
  const [highlightIndex, setHighlightIndex] = useState<number>(-1);
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const [overlayTop, setOverlayTop] = useState<number>(0);

  // For Toggle Option
  const _handleToggleOption = useCallback(() => {
    setIsOpen((prev) => {
      if (!prev) _onHighlightItem(0);
      return !prev;
    });
  }, []);

  // On Option Selected
  const _onOptionSelect = useCallback(
    (option: T, optionIndex: number) => {
      if (onSelect) {
        onSelect(option, optionIndex);
      }

      setIsOpen(false);
      setSelectedIndex(optionIndex);
    },
    [onSelect]
  );

  // Key Extractor for Each Keys to render optimize of react components
  const _keyExtractor = useCallback(
    (option: T) => {
      if (keyExtractor) {
        return keyExtractor(option);
      }
      // Default Key Extractor
      return `${Math.random()}`;
    },
    [keyExtractor]
  );

  // Label Extractor to Correct Label Text
  const _labelExtractor = useCallback(
    (option: T) => {
      if (labelExtractor) {
        return labelExtractor(option);
      }
      // Default Value Extractor
      return `Need Label Extractor`;
    },
    [labelExtractor]
  );

  // Handling Keydown to Show Option
  const _onButtonKeyDown = useCallback(
    (event: KeyboardEvent) => {
      event.preventDefault();
      const { SPACE, ENTER, DOWN_ARROW } = KeyboardKeys;

      if (SPACE === event.key) {
        setIsOpen((prev) => {
          if (!prev) _onHighlightItem(0);
          return !prev;
        });
        return;
      }

      const isIncludeKey = Object.values([ENTER, DOWN_ARROW]).some(
        (key) => key === event.key
      );

      if (isIncludeKey) {
        setIsOpen(true);
        _onHighlightItem(0);
      }
    },
    [KeyboardKeys]
  );

  // When User MouseEnter or Keydown for each item
  const _onHighlightItem = useCallback((optionIndex: number) => {
    setHighlightIndex(optionIndex);
    if (optionIndex === -1) return;

    const curRef = optionRefs[optionIndex];
    if (curRef && curRef.current) {
      curRef.current.focus();
    }
  }, []);

  const _getNextOptionIndex = useCallback((index: number, options: T[]) => {
    const nextIndex =
      index === -1 || index === options.length - 1 ? 0 : index + 1;
    return nextIndex;
  }, []);

  const _getPreviousOptionIndex = useCallback((index: number, options: T[]) => {
    const previousIndex =
      index === -1 || index === 0 ? options.length - 1 : index - 1;
    return previousIndex;
  }, []);

  // When User Keydown for each item
  const _onOptionKeyDown = useCallback(
    (event: KeyboardEvent) => {
      event.preventDefault();
      if (highlightIndex === -1) return;
      const { ESC } = KeyboardKeys;
      if (ESC === event.key) {
        setIsOpen(false);
        return;
      }

      const { DOWN_ARROW, UP_ARROW, ENTER } = KeyboardKeys;

      const isDownArrow = DOWN_ARROW === event.key;
      const isUpArrow = UP_ARROW === event.key;
      const isEnter = ENTER === event.key;
      if (isDownArrow) {
        _onHighlightItem(_getNextOptionIndex(highlightIndex, options));
      }

      if (isUpArrow) {
        _onHighlightItem(_getPreviousOptionIndex(highlightIndex, options));
      }

      if (isEnter) {
        _onOptionSelect(options[highlightIndex], highlightIndex);
      }
    },
    [
      options?.length,
      KeyboardKeys,
      highlightIndex,
      _onHighlightItem,
      _getNextOptionIndex,
      _getPreviousOptionIndex,
      _onOptionSelect,
    ]
  );

  // Selected Label
  const selectedLabel = useMemo<string | number>(() => {
    return selectedIndex === -1
      ? label
      : labelExtractor
      ? labelExtractor(options[selectedIndex])
      : "Needed LabelExtractor to show correct";
  }, [selectedIndex, label, labelExtractor]);

  // Caret Classes for open or close
  const caretClasses = useMemo<string>(() => {
    const classes = "wo-widht-xs wo-height-xs wo-select__caret ";
    return isOpen
      ? classes + "wo-select__caret--open"
      : classes + "wo-select__caret--close";
  }, [isOpen]);

  // Handling Offset Of Selector
  useEffect(() => {
    setOverlayTop(
      labelRef.current?.offsetHeight ? labelRef.current?.offsetHeight + 2 : 0
    );
  }, [labelRef.current?.offsetHeight]);

  // Controlling Each Option Ref
  useEffect(() => {
    setOptionRefs(options.map((_) => createRef<HTMLLIElement>()));
  }, [options?.length]);

  // When Open and Ref focused to highligh option
  useEffect(() => {
    if (highlightIndex !== -1 && isOpen) {
      const ref = optionRefs[highlightIndex];
      if (ref && ref.current) {
        ref.current?.focus();
      }
    }
  }, [isOpen, highlightIndex]);

  // Handling Mouse Away Click Listener
  useEffect(() => {
    // Target Error Handler
    function assertIsNode(e: EventTarget | null): asserts e is Node {
      if (!e || !("nodeType" in e)) {
        throw new Error(`Node expected in Select Component`);
      }
    }

    let handleClickOutside = ({ target }: MouseEvent) => {
      if (selectContainerRef?.current && target) {
        // In a utility library:
        assertIsNode(target);

        // Click Outside of List
        if (
          !selectContainerRef?.current?.contains(target) &&
          !listContainerRef?.current?.contains(target) &&
          isOpen
        ) {
          setIsOpen(false);
        }
      }
    };

    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, [isOpen]);

  return (
    <div className="wo-select" ref={selectContainerRef}>
      <button
        ref={labelRef}
        className="wo-select__label"
        onClick={_handleToggleOption}
        onKeyDown={_onButtonKeyDown}
        aria-haspopup={true}
        aria-expanded={isOpen ? true : undefined}
        data-testid="WinOoSelectorButton"
      >
        <Text>{selectedLabel}</Text>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          fill="none"
          viewBox="0 0 24 24"
          strokeWidth={1.5}
          stroke="currentColor"
          className={caretClasses}
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          />
        </svg>
      </button>

      {isOpen && (
        <ul
          role="menu"
          ref={listContainerRef}
          aria-hidden={isOpen ? undefined : "false"}
          className="wo-select__overlay"
          style={{ top: overlayTop }}
        >
          {options?.map((option, optionIndex) => {
            const ref = optionRefs[optionIndex];
            const isSelected = selectedIndex === optionIndex;
            const isHighlighted = highlightIndex === optionIndex;
            // Render Options can read from parent scope with the function called renderOption
            const renderOptionProps = {
              item: option,
              isSelected,
              isHighlighted,
              classes: `wo-select__option 
                ${isSelected ? "wo-select__option--selected" : ""}
                ${isHighlighted ? "wo-select__option--highlighted" : ""}
              `,
            };

            // If user has used renderOption
            // We'll pass with the default classes
            // And Return User Component
            if (renderOption) {
              return (
                <li
                  role="menuitemradio"
                  ref={ref}
                  tabIndex={isHighlighted ? -1 : 0}
                  key={_keyExtractor(option)}
                  onClick={() => _onOptionSelect(option, optionIndex)}
                  onMouseEnter={() => _onHighlightItem(optionIndex)}
                  onMouseLeave={() => _onHighlightItem(-1)}
                  onKeyDown={(e) => _onOptionKeyDown(e)}
                >
                  {renderOption(renderOptionProps)}
                </li>
              );
            }

            // Else Our Default Selecter Style
            return (
              <li
                role="menuitemradio"
                ref={ref}
                tabIndex={isHighlighted ? -1 : 0}
                key={_keyExtractor(option)}
                onClick={() => _onOptionSelect(option, optionIndex)}
                onMouseEnter={() => _onHighlightItem(optionIndex)}
                onMouseLeave={() => _onHighlightItem(-1)}
                onKeyDown={(e) => _onOptionKeyDown(e)}
                className={renderOptionProps.classes}
              >
                <Text>{_labelExtractor(option)}</Text>
                {isSelected && (
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    fill="none"
                    viewBox="0 0 24 24"
                    strokeWidth="1.5"
                    stroke="currentColor"
                    className="wo-widht-xs wo-height-xs"
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      d="M11.48 3.499a.562.562 0 011.04 0l2.125 5.111a.563.563 0 00.475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 00-.182.557l1.285 5.385a.562.562 0 01-.84.61l-4.725-2.885a.563.563 0 00-.586 0L6.982 20.54a.562.562 0 01-.84-.61l1.285-5.386a.562.562 0 00-.182-.557l-4.204-3.602a.563.563 0 01.321-.988l5.518-.442a.563.563 0 00.475-.345L11.48 3.5z"
                    />
                  </svg>
                )}
              </li>
            );
          })}
        </ul>
      )}
    </div>
  );
};

export default Select;
