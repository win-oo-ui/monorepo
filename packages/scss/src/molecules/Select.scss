@import "foundation/all";

.wo-select {
  display: flex;
  flex-direction: column;
  margin: 0;
  position: relative;
}

.wo-select__label {
  @include padding("xxs");
  background-color: $form-bg-color;
  border: 1px solid $form-border-color;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  font-family: $body-font-family;
  justify-content: space-between;
  position: relative;
}

.wo-select__caret {
  transition: all 0.25s ease-in-out;
}

.wo-select__caret--open {
  transform: rotate(180deg);
}

.wo-select__caret--close {
  transform: rotate(0deg);
}

.wo-select__overlay {
  padding: 0;
  margin: 0;
  background-color: $form-bg-color;
  border: 1px solid $form-border-color;
  position: absolute;
  width: 100%;
  z-index: 99;
  list-style-type: none;
}

.wo-select__option {
  @include padding("xs");
  align-items: center;
  cursor: pointer;
  display: flex;
  justify-content: space-between;
  position: relative;
}

.wo-select__option--selected {
  background-color: $form-selected-bg-color;
}

.wo-select__option:hover {
  background-color: $form-border-color;
}
