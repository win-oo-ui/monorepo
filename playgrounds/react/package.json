{
  "name": "@winoo/playground-react",
  "version": "0.0.0",
  "main": "index.js",
  "license": "MIT",
  "files": [
    "lib",
    "src"
  ],
  "scripts": {
    "dev": "parcel src/index.html -p 3000"
  },
  "devDependencies": {
    "@types/react": "^18.0.28",
    "@types/react-dom": "^18.0.11",
    "parcel-bundler": "^1.12.5",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "typescript": "^4.9.5"
  },
  "private": true,
  "publishConfig": {
    "access": "restricted"
  }
}
