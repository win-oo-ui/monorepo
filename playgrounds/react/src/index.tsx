import React from "react";
import { createRoot } from "react-dom/client";
import { Dot, Text, Box, Select } from "@winoo/react";
import { FontSize, Spacing } from "@winoo/foundation";
// Importing All Css
import "@winoo/scss/lib/Select.css";
import "@winoo/scss/lib/Button.css";
import "@winoo/scss/lib/Text.css";
import "@winoo/scss/lib/Dot.css";
import "@winoo/scss/lib/Utilities.css";
import "@winoo/scss/lib/global.css";

const container = document.getElementById("app-root");
const root = createRoot(container!);

const mockData: { id: string; item_name: string }[] = [
  {
    id: "1",
    item_name: "Data base",
  },
  {
    id: "2",
    item_name: "Data base2",
  },
  {
    id: "3",
    item_name: "Hello World",
  },
];

const App = () => {
  return (
    <>
      <Box margin={{ value: Spacing.lg }}>
        <Text size={FontSize.base}>Hello</Text>
        <Dot hexCode="#ff0000" size={Spacing.xxxs} />
        <Dot hexCode="#ff0000" size={Spacing.xxs} />
        <Dot hexCode="#ff0000" size={Spacing.xs} />
        <Dot hexCode="#ff0000" size={Spacing.sm} />
        <Dot hexCode="#ff0000" size={Spacing.md} />
        <Dot hexCode="black" size={Spacing.lg} />
        <Dot hexCode="black" size={Spacing.xl} />
        <Dot hexCode="black" size={Spacing.xxl} />
        <Dot hexCode="black" size={Spacing.xxxl} />
      </Box>
      <Box margin={{ value: Spacing.lg }}>
        <Select
          options={mockData}
          keyExtractor={(item) => item.id}
          labelExtractor={(item) => item.item_name}
        />
      </Box>
    </>
  );
};

root.render(<App />);
