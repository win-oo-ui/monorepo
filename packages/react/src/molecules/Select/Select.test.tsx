import React from "react";
import Select from "./Select";

import { fireEvent, render } from "@testing-library/react";

const mockData = [
  {
    id: "1",
    label: "Mg Mg",
    value: "mg",
  },
  {
    id: "2",
    label: "Ag Ag",
    value: "ag",
  },
  {
    id: "3",
    label: "Lwin",
    value: "lmp",
  },
];

// 1. Get All Length Of MockData to show correctly list
test("Render: All SelectorMock Data passed to it:/Molecules/Select", () => {
  const { getAllByRole, getByTestId } = render(
    <Select
      label="Hello World"
      options={mockData}
      labelExtractor={(item) => item.label}
    />
  );

  // First we need to open selector
  fireEvent.click(getByTestId("WinOoSelectorButton"));

  // Mokck Data Length must be same as Test Data Length
  expect(getAllByRole("menuitemradio")).toHaveLength(mockData.length);
});

// 2. If we use our custom Children List, We need to check parent data props
test("Can have pass valid Props with renderOption from Component:/Molecules/Select", () => {
  const { getAllByTestId, getByTestId } = render(
    <Select
      label="Hello World"
      options={mockData}
      labelExtractor={(item) => item.label}
      renderOption={({ item }) => {
        return <div data-testid="TestComponent">{item.label}</div>;
      }}
    />
  );

  // First we need to open selector
  fireEvent.click(getByTestId("WinOoSelectorButton"));

  // Mokck Data Length must be same as Custom Test Data Length
  expect(getAllByTestId("TestComponent")).toHaveLength(mockData.length);
});

// 3. When we click each item...
test("Can Get Correct Data when User Selected Item:/Molecules/Select", () => {
  const jestTestFunction = jest.fn();

  const { getAllByRole, getByTestId } = render(
    <Select
      label="Hello World"
      options={mockData}
      labelExtractor={(item) => item.label}
      onSelect={jestTestFunction}
    />
  );

  // First we need to open selector
  fireEvent.click(getByTestId("WinOoSelectorButton"));
  // Select First Row of items , index number => 0
  const firsSelectBox = getAllByRole("menuitemradio")[0];
  fireEvent.click(firsSelectBox);

  // JEST function will return as the same index[0] mock data and Index number Zero
  expect(jestTestFunction).toHaveBeenCalledWith(mockData[0], 0);
});

// 4. When we select item, Must change Selected Item to Select Box
test("Can Get Correct Selected Text Appear when User Selected Item:/Molecules/Select", () => {
  const { getAllByRole, getByTestId } = render(
    <Select
      label="Hello World"
      options={mockData}
      labelExtractor={(item) => item.label}
    />
  );

  // First we need to open selector
  fireEvent.click(getByTestId("WinOoSelectorButton"));
  // Select First Row of items , index number => 0
  const firsSelectBox = getAllByRole("menuitemradio")[0];
  fireEvent.click(firsSelectBox);

  // JEST function will return as the same index[0] mock data and Index number Zero
  expect(getByTestId("WinOoSelectorButton")).toHaveTextContent(
    mockData[0].label
  );
});

// 5. SnapShop testing...
test("SnapShop: Selector Mock Data:/Molecules/Select", () => {
  const { getAllByRole, getByTestId, asFragment } = render(
    <Select
      label="Hello World"
      options={mockData}
      labelExtractor={(item) => item.label}
    />
  );

  // First we need to open selector
  fireEvent.click(getByTestId("WinOoSelectorButton"));
  // Select First Row of items , index number => 0
  const firsSelectBox = getAllByRole("menuitemradio")[0];
  fireEvent.click(firsSelectBox);
  // JEST function will return as the same index[0] mock data and Index number Zero
  expect(asFragment()).toMatchSnapshot();
});

// 6. SnapShop Base testing...
test("SnapShop: Selector Base:/Molecules/Select", () => {
  const { asFragment } = render(
    <Select
      label="Base Framgment"
      options={mockData}
      labelExtractor={(item) => item.label}
    />
  );

  // JEST function will return as the same index[0] mock data and Index number Zero
  expect(asFragment()).toMatchSnapshot();
});

// 7. SnapShop Open State testing...
test("SnapShop: Selector Open State:/Molecules/Select", () => {
  const { asFragment, getByTestId } = render(
    <Select
      label="Open State Framgment"
      options={mockData}
      labelExtractor={(item) => item.label}
    />
  );

  // First we need to open selector
  fireEvent.click(getByTestId("WinOoSelectorButton"));

  // JEST function will return as the same index[0] mock data and Index number Zero
  expect(asFragment()).toMatchSnapshot();
});

// 8. Can Customize Selector Label
test("Can Customize Select Label in Atomic:Select:/Molecules/Select", () => {
  const { getByText } = render(
    <Select
      label="CUSTOM SELECT LABEL"
      options={mockData}
      labelExtractor={(item) => item.label}
    />
  );

  expect(getByText(/CUSTOM SELECT LABEL/)).toBeInTheDocument();
});
