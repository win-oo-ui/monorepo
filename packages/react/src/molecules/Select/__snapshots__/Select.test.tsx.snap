// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShop: Selector Base:/Molecules/Select 1`] = `
<DocumentFragment>
  <div
    class="wo-select"
  >
    <button
      aria-haspopup="true"
      class="wo-select__label"
      data-testid="WinOoSelectorButton"
    >
      <span
        class="wo-text-base"
      >
        Base Framgment
      </span>
      <svg
        class="wo-widht-xs wo-height-xs wo-select__caret wo-select__caret--close"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`SnapShop: Selector Mock Data:/Molecules/Select 1`] = `
<DocumentFragment>
  <div
    class="wo-select"
  >
    <button
      aria-haspopup="true"
      class="wo-select__label"
      data-testid="WinOoSelectorButton"
    >
      <span
        class="wo-text-base"
      >
        Mg Mg
      </span>
      <svg
        class="wo-widht-xs wo-height-xs wo-select__caret wo-select__caret--close"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`SnapShop: Selector Open State:/Molecules/Select 1`] = `
<DocumentFragment>
  <div
    class="wo-select"
  >
    <button
      aria-expanded="true"
      aria-haspopup="true"
      class="wo-select__label"
      data-testid="WinOoSelectorButton"
    >
      <span
        class="wo-text-base"
      >
        Open State Framgment
      </span>
      <svg
        class="wo-widht-xs wo-height-xs wo-select__caret wo-select__caret--open"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 8.25l-7.5 7.5-7.5-7.5"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </button>
    <ul
      class="wo-select__overlay"
      role="menu"
      style="top: 0px;"
    >
      <li
        class="wo-select__option 
                
                wo-select__option--highlighted
              "
        role="menuitemradio"
        tabindex="-1"
      >
        <span
          class="wo-text-base"
        >
          Mg Mg
        </span>
      </li>
      <li
        class="wo-select__option 
                
                
              "
        role="menuitemradio"
        tabindex="0"
      >
        <span
          class="wo-text-base"
        >
          Ag Ag
        </span>
      </li>
      <li
        class="wo-select__option 
                
                
              "
        role="menuitemradio"
        tabindex="0"
      >
        <span
          class="wo-text-base"
        >
          Lwin
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
